context.go

```diff
@@ -0,0 +1,239 @@
+package huma
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/danielgtaylor/huma/negotiation"
+	"github.com/fxamacker/cbor/v2"
+	"github.com/goccy/go-yaml"
+)
+
+// ContextFromRequest returns a Huma context for a request, useful for
+// accessing high-level convenience functions from e.g. middleware.
+func ContextFromRequest(w http.ResponseWriter, r *http.Request) Context {
+	return &hcontext{
+		Context:        r.Context(),
+		ResponseWriter: w,
+		r:              r,
+	}
+}
+
+// Context provides a request context and response writer with convenience
+// functions for error and model marshaling in handler functions.
+type Context interface {
+	context.Context
+	http.ResponseWriter
+
+	// AddError adds a new error to the list of errors for this request.
+	AddError(err error)
+
+	// HasError returns true if at least one error has been added to the context.
+	HasError() bool
+
+	// WriteError writes out an HTTP status code, friendly error message, and
+	// optionally a set of error details set with `AddError` and/or passed in.
+	WriteError(status int, message string, errors ...error)
+
+	// WriteModel writes out an HTTP status code and marshalled model based on
+	// content negotiation (e.g. JSON or CBOR). This must match the registered
+	// response status code & type.
+	WriteModel(status int, model interface{})
+}
+
+type hcontext struct {
```

Comment:
This isn't a very informative type name; what is meant by `hcontext`?

Reply:
The name `context` is taken by a standard library package, so we append the name of this package. I guess that means it stands for "Huma context". It was just a way to prevent shadowing and is an internal implementation detail of the public `huma.Context` interface.